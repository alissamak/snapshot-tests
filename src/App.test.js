import { render, screen, waitFor } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';
import GitHubCard from './components/GitHubCard';

test('renders a snapshot', () => {
    const tree = renderer.create(<GitHubCard/>).toJSON()
    expect(tree).toMatchSnapshot()
    
});

// test('renders a snapshot', () => {
//   const tree = renderer.create(<App/>).toJSON()
//   expect(tree).toMatchSnapshot()
  
// });

beforeEach(()=> {
    //sorts eveyrthing back to initial state before each test
    fetch.resetMocks();
})

test('fetch gitHub profile name from GitHub REST API using jest fetch mock', async () => {
    fetch.mockResponseOnce(JSON.stringify({name: 'Alissa M'}))
    render(<App/>)
    const gitHubName = await waitFor(() => screen.getByRole('Card.Title'))
    expect(gitHubName).toHaveTextContent('Alissa M')
})

test('check correct URL is fetched from Github REST API with jest fetch mock', async () => {
    fetch.mockResponseOnce(JSON.stringify({html_url:'https://api.github.com/users/alissamak'}))
    render(<App/>)
    const gitHubURL = await waitFor(() => screen.getByRole('link'))
    expect(gitHubURL).toHaveAttribute('href', expect.stringContaining('github.com'))
})

//bonus
test('check if github profile image has correct src sttribute fetched from Github REST API', async () => {
    fetch.mockResponseOnce(JSON.stringify({avatar_url:'https://avatars.githubusercontent.com/u/96849455?v=4'}))
    render(<App/>)
    const avatarURL = await waitFor(() => screen.getByAltText('Github profile image'))
    expect(avatarURL).toHaveAttribute('src', expect.stringContaining('githubusercontent'))
})