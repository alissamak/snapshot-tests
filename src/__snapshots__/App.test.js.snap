// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App pt-5"
>
  <div
    className="col"
  >
    <div
      className="row"
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <div
        className="card"
        style={
          Object {
            "border": "none",
          }
        }
      >
        <div
          className="card"
          style={
            Object {
              "margin": "auto",
              "width": "18rem",
            }
          }
        >
          <img
            alt="Github profile image"
            className="card-img-top"
            height="200px"
            src=""
            width="200px"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            />
            <p
              className="card-text"
            />
            <p
              className="card-text"
            >
              <a
                href=""
              >
                <button
                  className="btn btn-primary"
                  disabled={false}
                  type="button"
                >
                  Link to GitHub Profile
                </button>
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
